cpp/src/io/json/read_json.cu

```diff
@@ -278,25 +307,73 @@ table_with_metadata read_json(host_span<std::unique_ptr<datasource>> sources,
                  "Multiple inputs are supported only for JSON Lines format");
   }
 
-  datasource::owning_buffer<rmm::device_uvector<char>> bufview =
-    get_record_range_raw_input(sources, reader_opts, stream);
+  std::for_each(sources.begin(), sources.end(), [](const auto& source) {
+    CUDF_EXPECTS(source->size() < std::numeric_limits<int>::max(),
+                 "The size of each source file must be less than INT_MAX bytes");
+  });
 
-  // If input JSON buffer has single quotes and option to normalize single quotes is enabled,
-  // invoke pre-processing FST
-  if (reader_opts.is_enabled_normalize_single_quotes()) {
-    normalize_single_quotes(bufview, stream, rmm::mr::get_current_device_resource());
+  size_t const batch_size_ub = std::numeric_limits<int>::max();
+  size_t const chunk_offset  = reader_opts.get_byte_range_offset();
+  size_t chunk_size          = reader_opts.get_byte_range_size();
+  chunk_size                 = !chunk_size ? sources_size(sources, 0, 0) : chunk_size;
+
+  size_t start_source = 0, cur_size = 0;
+  for (size_t sum = 0; start_source < sources.size() && sum <= chunk_offset;
+       sum += sources[start_source]->size(), start_source++)
+    ;
+  start_source--;
+  std::vector<size_t> batch_positions, batch_sizes;
```

Comment:
Not sure if this is in the dev guide, but we avoid multiple declarations in a single line
```suggestion
  std::vector<size_t> batch_positions;
  std::vector<size_t> batch_sizes;
```